class_diagrams/pedometer.md

```diff
@@ -0,0 +1,105 @@
+![](http://www.plantuml.com/plantuml/png/jLHHRyCq47xthpX5IBJ0EY7Ueb1jQykenDXek_7i9PVMQc87-z9IJVntpaaM1jJjYaGUskJytVVTPv_lnf6qL9U585C16NWrHAQ4J5bCIHajX4C2BKjqbKnnYPMqaeo5E1PkgxJ_BC5jP6RsIgyXbuN33ojf8Ps3Dm7ytCbqkxQcrjd45COcYSMiJj6En_2ZqNYOuluuJZQbCEpN3KewZd6edV5KAqmtccFhmt7WJk6UOOystQZiE0JpdDqpwDuKMKYkO4WZ-9FSeSxGYZ-4-FBxlnx8C5TQ-TAT4Er6H1DZ4MwzDd5-RJ6EOSay646Iv5QMk3TsowoyPLydoSIY9EJF1ZFeDinn3bAowBPlAE3A4UE-Y1zV7TgTV2qmMPBFvALszzpV5c_Rs9X5CkKgIUbcqE9YQDU6ySweZDTRfuDtmopTlGtZHjJRximzOMPA9BHiQgxHafB50qh7DalKzAmLmM3-y3oCe3-mrgdHSruSzA96Jz6HDOSk518fwwaa6JKSCasnG2jR5ozDZTznt1hgOh_KalCqltDuWpBZZeqXBuma3X2MbK_dQhjWM7NH9aVGDsVhWmuLnX5Rl5xCRgVpd-yZlyDSOjoL6g1S4bOhkKKTO7oQJHyVPglPudgvcZr50POf-gRoN3zOKxAYP-kR-UrCBmH8-_SNXVi8-WlaYxDDumVFgBUshCjIseDlIUL8fQvhtGZw8T0p-gz1qra-b_--WNzsPXiFw4raOQn2TrTRZTd_9zGQwnr5y74eU0Ny7TEQSC0qxjFrdgVvrvWfbDuwSVBCHgFJfSSmXjzgz6M0OsM4lQBDUSwmZ__8MOzEZaZFkKPouAyKvDQKGbmqTKRd-T_d8Jh07MeM8tEfI71UU05JKrMJ4o84WkZRwAkhA_uzFkt4sKiVhg3rc1hjwXBTPvY2WohivVILFyTPC1wrWznOSVfcNym7Gfmw_b4s6f-Rg2S94cQuOFfTnXNmBsvGPtLP_0a0)
+
+
+# Pedometer
+The Pedometer measures the number of steps taken by using data from the IMU. As it requires a window of data to operate on and not a single reading we feed it into an effect class. This effect class can be used to do a simple detection of a step by finding the local maxima of the acceleration magnitude data. Only peaks with a minimum height above one standard deviation are treated as a step. [1] It may also be used with an AI algorithm for more accurate detection of steps or to detect push (and push type) if the user is in a wheelchair. [2] Based on the number of steps we can do a simple calculation for the number of calories burned (CaloriesBurnedReducer)[3][4].
```

Comment:
In other work done so far, we have mentioned Accelerometer, not Pedometer or IMU. 

Just a note so that we remember to use the same names later

Reply:
Pedometer is just the application name. We can call it a step meter instead. It's not a separate sensor it uses accelerometer data under the hood. Most applications wile they call it an accelerometer it's actually an IMU. The accelerometer data is only one part of its output. They're so cheap these days that its rare to find an accelerometer by itself.

Reply:
That's interesting, because in the requirements, we were trying to define with Fulvio or not the boundary for acceleration minimum detection or pedometer trigger. If we can clarify it along the way, I think it is useful to mention both, at the sensing device level, and the application module.